microsetta_interface/static/vue_survey_form.js

```diff
@@ -37,10 +37,244 @@ var vm = new Vue({
+                            if(checkboxLabelElement.children[k].nodeName == "INPUT") {
+                                checkboxLabelElement.children[k].checked = false;
+                            }
+                        }
+                        if(checkboxLabelElement.classList.contains("is-checked")) {
+                            checkboxLabelElement.classList.toggle("is-checked");
+                        }
+                    }
+                }
+            }
+        }
+    } else if(group.classList.contains("field-input") ) {
+        for (let i = 0; i < group.childNodes.length; i++) {
+            let theNode = group.childNodes[i];
+            if (theNode.nodeName == "DIV" && theNode.classList.contains("field-wrap")) {
+                let wrapper = theNode.children[0];
+                let inputElement = wrapper.children[0];
+
+                if(newState == "display") {
+                    theNode.style.display = "";
+                    inputElement.value = "";
+                    vm.methodToUpdate(inputElement.name, false, false);
+                } else {
+                    theNode.style.display = "none";
+                    inputElement.value = "Unspecified";
+                    vm.methodToUpdate(inputElement.name, true, false);
+                }
+            }
+        }
+    } else if(group.classList.contains("field-textArea")) {
+        for (let i = 0; i < group.childNodes.length; i++) {
+            let theNode = group.childNodes[i];
+            if (theNode.nodeName == "DIV" && theNode.classList.contains("field-wrap")) {
+                let inputElement = theNode.children[0];
+
+                if(newState == "display") {
+                    theNode.style.display = "";
+                    inputElement.innerText = "";
+                    vm.methodToUpdate(inputElement.name, false, false);
+                } else {
+                    theNode.style.display = "none";
+                    inputElement.innerText = "Unspecified";
+                    vm.methodToUpdate(inputElement.name, true, false);
+                }
+            }
+        }
+    } else if(group.classList.contains("field-select")) {
+        for(let i = 0; i < group.childNodes.length; i++) {
+            let theNode = group.childNodes[i];
+            if (theNode.nodeName == "DIV" && theNode.classList.contains("field-wrap")) {
+                let inputElement = theNode.children[0];
+
+                if(newState == "display") {
+                    theNode.style.display = "";
+                    inputElement.selectedIndex = -1;
+                    vm.methodToUpdate(inputElement.name, false, false);
+                } else {
+                    theNode.style.display = "none";
+                    inputElement.selectedIndex = 0;
+                    vm.methodToUpdate(inputElement.name, true, false);
+                }
+            }
+        }
+    } else {
+        // Why am I here?
```

Comment:
We can't do anything meaningful with a client-side throw so I'm not sure it has any value. That said, I cleaned up the comment because it's superfluous.

Reply:
Good point, and thanks